draft-ietf-quic-transport.md

```diff
@@ -2714,13 +2714,19 @@ but not the UDP or IP header.
 
 The initial client packet SHOULD be padded to exactly 1200 octets unless the
 client has a reasonable assurance that the PMTU is larger.  Sending a packet of
-this size ensures that the network path supports an MTU of this size and helps
-reduce the amplitude of amplification attacks caused by server responses toward
-an unverified client address.
+this size ensures that the network path from the client to the server supports
+an MTU of this size and helps reduce the amplitude of amplification attacks
+caused by server responses toward an unverified client address.
 
 Servers MUST ignore an initial plaintext packet from a client if its total size
 is less than 1200 octets.
 
+Similarly, servers MUST ensure that the first handshake packet they send to
+clients, and any retransmissions of those octets, has a QUIC packet size that is
+the same as the received initial client packet, unless the server knows the PMTU
+to the client to be smaller. Sending a packet of this size ensures that the
```

Comment:
This "smaller" clause here means that the server-to-client path could end up with a PMTU less than 1200.  I think that you need to set a minimum of 1200 here explicitly.

Reply:
The client side "at least 1200 byte" is justified by a security concern, but that concern does not apply to server responses. I think we need some better text, and that the justification for the padding should come after the requirement of minimum 1280 MTU. Setting a minimum size for the first handshake packet ensures that the handshake will fail if the network does not support the recommended MTU, and is only justified if we want enforce the "minimum 1280" requirement.

But if we do that, maybe we should develop the idea in a full paragraph. For example, is it OK to meet the requirement in IPv4 by setting the DF bit to "fragmentation authorized"?

Reply:
I am a bit skeptical about the requirement for "any retransmissions of those octets". It will be yet another special case in the handling of re-transmission. The net effect will be that servers pad all handshake message re-transmissions to 1200 bytes.

Reply:
I agree that this would make it much more difficult to implement.  You need special handling for this case and I can see how you would end up with a fixed size for Handshake packets.  That would be unfortunate for several reasons.